packages/server/graphql/private/mutations/removeAuthIdentity.ts

```diff
@@ -0,0 +1,80 @@
+import base64url from 'base64url'
+import bcrypt from 'bcrypt'
+import crypto from 'crypto'
+import {AuthIdentityTypeEnum, Security} from 'parabol-client/types/constEnums'
+import {r} from 'rethinkdb-ts'
+import util from 'util'
+import AuthIdentityLocal from '../../../database/types/AuthIdentityLocal'
+import PasswordResetRequest from '../../../database/types/PasswordResetRequest'
+import getMailManager from '../../../email/getMailManager'
+import resetPasswordEmailCreator from '../../../email/resetPasswordEmailCreator'
+import generateRandomString from '../../../generateRandomString'
+import getUsersByDomain from '../../../postgres/queries/getUsersByDomain'
+import updateUser from '../../../postgres/queries/updateUser'
+import {MutationResolvers} from '../../private/resolverTypes'
+
+const randomBytes = util.promisify(crypto.randomBytes)
+
+const removeAuthIdentity: MutationResolvers['removeAuthIdentity'] = async (
+  _source,
+  {domain, identityType, addLocal},
+  {ip}
+) => {
+  // VALIDATION
+  const users = await getUsersByDomain(domain)
+  if (!users.length) {
+    return {error: {message: 'No user emails match that domain'}}
+  }
+
+  // RESOLUTION
+  const updateUsersPromises = users.map(async (user) => {
+    const {id: userId, identities} = user
+    const filteredIdentities = identities.filter((identity) => identity.type !== identityType)
```

Comment:
-1 what if they already have a local & google identity & `addLocal` is true?
business logic should be to add local only if local doesn't already exist

Reply:
If the user only has a Google account and `addLocal` is false, should we leave identities empty or create a new local one? Leaving it empty would mean they can't use forget password or login

Reply:
i sure how they set `addLocal: true` 😬 
i think that's OK. the only use case for `addLocal: false` is if they move to SSO & don't want people to login via gmail or password. Or if we add a new oauth2 strategy & we just want to remove the 1.